java/src/test/java/scenario/ScenarioSteps.java

```diff
@@ -74,6 +76,8 @@ public class ScenarioSteps {
     private static final Path DOCKER_COMPOSE_DIR = Paths.get(FIXTURES_DIR.toString(), "docker-compose")
             .toAbsolutePath();
     private static final String DEFAULT_LISTENER_NAME = "";
+    private static Checkpointer checkpointer;
```

Comment:
This should not be static as it should not persist across scenarios. The checkpointer should be associated with a specific EventListener instance. The EventListener should call the checkpointer when it is asked for the next event. Perhaps a specific CheckpointEventListener implementation class that looks like the current EventListener but also does the checkpointing.